subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/GradleBuildSmokeTest.groovy

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.smoketests
+
+
+import org.gradle.testkit.runner.TaskOutcome
+import org.gradle.util.GradleVersion
+
+class GradleBuildSmokeTest extends AbstractGradleceptionSmokeTest {
+
+    def "can build Gradle distribution"() {
+        when:
+        result = runner(':distributions-full:binDistributionZip', ':distributions-full:binInstallation', '--stacktrace')
```

Comment:
Should this run with `--warning-mode=all`?

Reply:
That's set deeper down in the runner here: https://github.com/gradle/gradle/blob/8193ff695e42a59fda0cbadcc49936b99b1e5535/subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy#L240
via here: https://github.com/gradle/gradle/blob/8193ff695e42a59fda0cbadcc49936b99b1e5535/subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy#L263-L268